web-ui/src/main/resources/catalog/components/viewer/ViewerDirective.js

```diff
@@ -110,9 +110,17 @@
               scope.zoom = function(map, delta) {
                 gnMap.zoom(map, delta);
               };
+              
               scope.zoomToMaxExtent = function(map) {
-                map.getView().fit(map.getView().
-                    getProjection().getExtent(), map.getSize());
+                var proj = map.getView().getProjection();
+                var extent = proj.getExtent();
+                if (gnViewerSettings.initialExtent && gnViewerSettings.mapConfig.projection && 
+                  gnViewerSettings.mapConfig.projection !== proj.getCode()) {
+                  // Use reprojected initial extent if it is defined
+                  extent = ol.proj.transformExtent(
+                    gnViewerSettings.initialExtent, gnViewerSettings.mapConfig.projection, proj, 8)
+                }
+                map.getView().fit(extent, map.getSize());
```

Comment:
If an initial extent was defined in the viewer settings and its projection is different from the current map projection, use and reproject the initial extent. This makes more sense when switching to EPSG:3857 for instance, where the projection extent usually encapsulates the whole world.